src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -380,8 +380,77 @@ HDF5IOHandlerImpl::extendDataset(Writable* writable,
 }
 
 void
-HDF5IOHandlerImpl::openFile(Writable* writable,
-                            Parameter< Operation::OPEN_FILE > const& parameters)
+HDF5IOHandlerImpl::availableChunks(
+    Writable * writable,
+    Parameter< Operation::AVAILABLE_CHUNKS > & parameters )
+{
+    auto res = m_fileIDs.find( writable );
+    hid_t dataset_id = H5Dopen(
+        res->second,
+        concrete_h5_file_position( writable ).c_str(),
+        H5P_DEFAULT );
+    VERIFY(
+        dataset_id >= 0,
+        "[HDF5] Internal error: Failed to open HDF5 dataset during dataset "
+        "read" );
+    hid_t dataset_space = H5Dget_space( dataset_id );
+    int ndims = H5Sget_simple_extent_ndims( dataset_space );
+    VERIFY(
+        ndims >= 0,
+        "[HDF5]: Internal error: Failed to retrieve dimensionality of "
+        "dataset "
+        "during dataset read." );
+
+    // now let's figure out whether this one has chunks
+    hid_t propertyList = H5Dget_create_plist( dataset_id );
+    std::vector< hsize_t > chunkExtent( ndims, 0 );
+    int chunkDimensionality =
+        H5Pget_chunk( propertyList, ndims, chunkExtent.data() );
+
+    if( chunkDimensionality < 0 )
+    {
+        std::cerr << R"END(
+[HDF5] Warning: HDF5 does not carry information on available chunks.
+       Returning instead the whole dataset as one chunk.
+       Some positions in the chunk may hold undefined data.
+)END" << std::endl;
+    }
+    else
+    {
+        // so, the dataset indeed has chunks
+        // alas, this backend doesn't write chunks, so for now, reading them is
+        // unimplemented
+        std::cerr << R"END(
+[HDF5] Warning: HDF5 dataset has chunked layout. Since the openPMD API does not
+       support writing chunked HDF5 datasets, the whole dataset will be
+       returned as one chunk.
+       Some positions in the chunk may hold undefined data.
+)END" << std::endl;
```

Comment:
> I don't understand this warning.
> A user reads a file from elsewhere with openPMD-api and we say: "we found chunks, but we don't tell you, since we cannot write chunks." That makes no sense to me.

Simple reason: I can't really test an implementation of that easily at the moment, since I would need to create a chunked dataset for that, which the openPMD API doesn't currently do. If the openPMD API doesn't support writing chunks yet, it's fair that reading them is unsupported as well?

> Some positions in the chunk may hold undefined data.

Undefined as in "not defined explicitly by the writer". In ADIOS, if the writer writes data only to a subset of the available positions in the dataset, `::availableChunks` will only return positions in the dataset that have been written to. In HDF5, no such checks are done. Returning the whole dataset as one chunks means that some positions in the chunk may not hold sensible data if the dataset has been written partially only.

Reply:
I think there is a confusion on what HDF5 chunks actually are - please see explanation above. They are not as flexible/useful as ADIOS blocks/PGs, but rather a low-level storage detail, i.e. for compression and (in size) extensible data sets.

Reply:
> I think there is a confusion on what HDF5 chunks actually are 

They are certainly different in meaning from chunks in ADIOS, which should be noted somewhere a user will see it.